bus-mapping/src/circuit_input_builder.rs

```diff
@@ -654,21 +727,30 @@ pub struct CircuitInputStateRef<'a> {
     pub tx: &'a mut Transaction,
     /// Transaction Context
     pub tx_ctx: &'a mut TransactionContext,
-    /// Step
-    pub step: &'a mut ExecStep,
 }
 
 impl<'a> CircuitInputStateRef<'a> {
+    /// Create a new step from a `GethExecStep`
+    pub fn new_step(&self, geth_step: &GethExecStep) -> Result<ExecStep, Error> {
```

Comment:
Related to my previous comment: maybe here we also need a new function to create a virtual step.